### ISSUE TITLE
evdev installation fails on fresh manjaro

### ISSUE THREAD

**Issue by owzim** [rbreaves/kinto#168]
**Describe the bug**

On install with these options:

* `1) Kinto - xkeysnail (udev/x11) - Recommended`
* `2) Mac Only & VMs on Macbooks`

```sh
Building wheels for collected packages: xkeysnail, evdev
  Building wheel for xkeysnail (setup.py) ... done
  Created wheel for xkeysnail: filename=xkeysnail-0.3.0-py3-none-any.whl size=17588 sha256=62e32d33c6414966553adbd72e369af515b01e72cf597a243b38c0a70d3d9db6
  Stored in directory: /tmp/pip-ephem-wheel-cache-162l5l1_/wheels/04/a6/12/65bb8ee26ce911fea9df2fd3ada5f9217099f6779f4c53e0fb
  Building wheel for evdev (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-gd9bpe30/evdev/setup.py'"'"'; __file__='"'"'/tmp/pip-install-gd9bpe30/evdev/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-tevbogr5
       cwd: /tmp/pip-install-gd9bpe30/evdev/
  Complete output (26 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.8
  creating build/lib.linux-x86_64-3.8/evdev
  copying evdev/util.py -> build/lib.linux-x86_64-3.8/evdev
  copying evdev/uinput.py -> build/lib.linux-x86_64-3.8/evdev
  copying evdev/genecodes.py -> build/lib.linux-x86_64-3.8/evdev
  copying evdev/ff.py -> build/lib.linux-x86_64-3.8/evdev
  copying evdev/evtest.py -> build/lib.linux-x86_64-3.8/evdev
  copying evdev/events.py -> build/lib.linux-x86_64-3.8/evdev
  copying evdev/eventio_async.py -> build/lib.linux-x86_64-3.8/evdev
  copying evdev/eventio.py -> build/lib.linux-x86_64-3.8/evdev
  copying evdev/ecodes.py -> build/lib.linux-x86_64-3.8/evdev
  copying evdev/device.py -> build/lib.linux-x86_64-3.8/evdev
  copying evdev/__init__.py -> build/lib.linux-x86_64-3.8/evdev
  running build_ext
  running build_ecodes
  writing ecodes.c (using /usr/include/linux/input.h /usr/include/linux/input-event-codes.h /usr/include/linux/uinput.h)
  building 'evdev._input' extension
  creating build/temp.linux-x86_64-3.8
  creating build/temp.linux-x86_64-3.8/evdev
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fPIC -I/usr/include/python3.8 -c evdev/input.c -o build/temp.linux-x86_64-3.8/evdev/input.o -std=c99 -Wno-error=declaration-after-statement
  unable to execute 'gcc': No such file or directory
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for evdev
  Running setup.py clean for evdev
Successfully built xkeysnail
Failed to build evdev
Installing collected packages: evdev, inotify-simple, xkeysnail
    Running setup.py install for evdev ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-gd9bpe30/evdev/setup.py'"'"'; __file__='"'"'/tmp/pip-install-gd9bpe30/evdev/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-iko7apxl/install-record.txt --single-version-externally-managed --compile --install-headers /usr/include/python3.8/evdev
         cwd: /tmp/pip-install-gd9bpe30/evdev/
    Complete output (25 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.8
    creating build/lib.linux-x86_64-3.8/evdev
    copying evdev/util.py -> build/lib.linux-x86_64-3.8/evdev
    copying evdev/uinput.py -> build/lib.linux-x86_64-3.8/evdev
    copying evdev/genecodes.py -> build/lib.linux-x86_64-3.8/evdev
    copying evdev/ff.py -> build/lib.linux-x86_64-3.8/evdev
    copying evdev/evtest.py -> build/lib.linux-x86_64-3.8/evdev
    copying evdev/events.py -> build/lib.linux-x86_64-3.8/evdev
    copying evdev/eventio_async.py -> build/lib.linux-x86_64-3.8/evdev
    copying evdev/eventio.py -> build/lib.linux-x86_64-3.8/evdev
    copying evdev/ecodes.py -> build/lib.linux-x86_64-3.8/evdev
    copying evdev/device.py -> build/lib.linux-x86_64-3.8/evdev
    copying evdev/__init__.py -> build/lib.linux-x86_64-3.8/evdev
    running build_ext
    ecodes.c already exists ... skipping build_ecodes
    building 'evdev._input' extension
    creating build/temp.linux-x86_64-3.8
    creating build/temp.linux-x86_64-3.8/evdev
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fPIC -I/usr/include/python3.8 -c evdev/input.c -o build/temp.linux-x86_64-3.8/evdev/input.o -std=c99 -Wno-error=declaration-after-statement
    unable to execute 'gcc': No such file or directory
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-gd9bpe30/evdev/setup.py'"'"'; __file__='"'"'/tmp/pip-install-gd9bpe30/evdev/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-iko7apxl/install-record.txt --single-version-externally-managed --compile --install-headers /usr/include/python3.8/evdev Check the logs for full command output.
Adding xhost fix...

Kinto install is complete.
```

**Install Type:** VM
**Distro:** Manjaro 19.0.2 (uptodate on stable)
**DE:** Gnome
**Branch:** master
**Commit:** 0c3e03a

**Additional context**
In another VM, same Manjaro but not so fresh, it does not fail.

**Comment by rbreaves** [comment]
Not entirely sure but is it indicating that gcc is not installed? If so you'd need that package to compile. Not sure why it would not setup or install the dependencies though, there's really no reference to needing gcc in the xkeysnail readme and nothing about kinto requires it unless you needed to compile kintox11 which I have never needed someone to do, the included binary has always worked for the older method.

**Comment by owzim** [comment]
On a fresh Manjaro GNOME 20.0.1 Minimal, this issue does not exist.

**Comment by rbreaves** [comment]
Interesting.. well I am install gcc right now on a fresh install to see if that makes xkeysnail work different/better during compile time and I can see why I wouldn't have caught that sooner. I virtualize Manjaro and due to less than stellar support for it as far ask Guest Additions for Vbox goes I have to install gcc and let it recompile itself for the kernel Manjaro uses, likely due to the rolling nature of kernels in Arch based distros.

**Comment by rbreaves** [comment]
@owzim Issue is indeed that you will need to install gcc for it to work under Manjaro, a prerequisite likely due to the rolling kernel and had gone unnoticed since I install gcc on that particular VM regardless of how much dev work I do in the VM, for reasons stated above.

I am sure this is likely the cause for the failure under 20.01 as well. #169 

I will be updating dev and master with the appropriate changes and will be closing these 2 tickets and possibly others if they are determined to be related.

**Comment by rbreaves** [comment]
There was a regression to this fix during the v1.2 update. It has now been merged once again in the latest update with additional fixes.

https://github.com/rbreaves/kinto/commit/9ce8612f54eb32f0ec3580d2ef07b8db3ed8ce15
